File: support_tools/swigwin-3.0.12/Examples/test-suite/python/li_std_set_runme.py
from li_std_set import *

s = set_string()

s.append("a")
s.append("b")
s.append("c")

sum = ""
for i in s:
    sum = sum + i

if sum != "abc":
    raise RuntimeError

i = s.__iter__()
if i.next() != "a":
    raise RuntimeError
if i.next() != "b":
    raise RuntimeError
if i.next() != "c":
    raise RuntimeError


b = s.begin()
e = s.end()
sum = ""
while (b != e):
    sum = sum + b.next()
if sum != "abc":
    raise RuntimeError

b = s.rbegin()
e = s.rend()
sum = ""
while (b != e):
    sum = sum + b.next()

if sum != "cba":
    raise RuntimeError


si = set_int()

si.append(1)
si.append(2)
si.append(3)
i = si.__iter__()

if i.next() != 1:
    raise RuntimeError
if i.next() != 2:
    raise RuntimeError
if i.next() != 3:
    raise RuntimeError


i = s.begin()
i.next()
s.erase(i)

b = s.begin()
e = s.end()
sum = ""
while (b != e):
    sum = sum + b.next()
if sum != "ac":
    raise RuntimeError


b = s.begin()
e = s.end()
if e - b != 2:
    raise RuntimeError

m = b + 1
if m.value() != "c":
    raise RuntimeError


s = pyset()
s.insert((1, 2))
s.insert(1)
s.insert("hello")


sum = ()
for i in s:
    sum = sum + (i,)

if (len(sum) != 3 or (not 1 in sum) or (not 'hello' in sum) or (not (1, 2) in sum)):
    raise RuntimeError

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/li_std_set_runme.rb
#!/usr/bin/env ruby
#
# Put script description here.
#
# 
# 
# 
#

require 'swig_assert'

require 'li_std_set'
include Li_std_set

swig_assert_each_line(<<'EOF', binding)
s = Set_string.new

s.push("a")
s.push("b")
s << "c"

sum = ''
s.each { |x| sum << x }
sum == 'abc'

b = s.begin  # only if swig iterators are on
e = s.end
sum = ''
while b != e; sum << b.value; b.next; end
sum == 'abc'

b = s.rbegin  # only if swig iterators are on
e = s.rend
sum = ''
while b != e; sum << b.value; b.next; end
sum == 'cba'


si = Set_int.new
si << 1
si.push(2)
si.push(3)

i = s.begin()
i.next()
s.erase(i)
s.to_s == 'ac'

b = s.begin
e = s.end
e - b == 2

m = b + 1
m.value == 'c'

s = LanguageSet.new
s.insert([1,2])
s.insert(1)
s.insert("hello")
#s.to_a == [1,[1,2],'hello']  # sort order: s.sort {|a,b| a.hash <=> b.hash}
# Test above is flawed as LanguageSet sorts by each element's hash, so the order will change from one invocation to the next. Sort a conversion to array instead.
sa = s.to_a.sort { |x, y| x.to_s <=> y.to_s }
sa == [1,[1,2],'hello']

EOF

iv = Set_int.new([0,1,2,3,4,5,6])
iv.delete_if { |x| x == 0 || x == 3 || x == 6 }
swig_assert_equal(iv.to_s, '1245', binding)

